### PR TITLE
adds details here and there

### PR DESCRIPTION
**Pull request by fstamour**
I added 1-2 small things I thought could be useful, then I went worked on the alist section just as the issue #180 suggested.

### CODE REVIEW

File: data-structures.md [hunk 1]
```diff
@@ -1354,9 +1372,20 @@ use `cdr` or `second` to get the value or even better `assoc-value list key` fro
 ;; (:FOO . "FOO")
 ~~~
 
-
 There is `assoc-if`, and `rassoc` to get a cons cell by its value.
 
+If the alist as repeating (duplicate) keys, you can use `remove-if-not`, for example, to retrieve all of them.
```
Comment (vindarel) [inline data-structures.md:1377]: typo: *has

File: iteration.md [hunk 2]
```diff
@@ -874,6 +874,13 @@ Use `nil` to ignore a term:
 ;; (X Y Z)
 ~~~
 
+Iterate over a list, 2 items at a time
+~~~lisp
+(loop for (key value) :on '(a 2 b 2 c 3) :by #'cddr
```
Comment (vindarel) [inline iteration.md:879]: Explain the strange-looking "cddr"?

**Comment by fstamour** [comment]
How about now?

**Comment by vindarel** [comment]
Nice additions, thank you!

**Pull request merged**